listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -46,6 +46,42 @@ export default class ListenCard extends React.Component<
   ListenCardProps,
   ListenCardState
 > {
+  static preciseTimestamp = (listen: Listen): string => {
+    const listened_at = listen.listened_at_iso || listen.listened_at * 1000;
+    const listenDate: Date = new Date(listened_at);
```

Comment:
I don't know how the JS Date object works here, but listened_at is going to be in the UTC timezone. Is this object in the local tz, or UTC, and how does it compare with the timezone in the next line?

Reply:
Date expects a UTC timestamp and toLocaleString will transform it according to the current TZ (unless `timeZone` is specified in the options).